### ISSUE TITLE
Generic name cannot be a function name

### ISSUE THREAD

**Issue by oovm** [intellij-rust/intellij-rust#10547]
## Environment

* **IntelliJ Rust plugin version:** 0.4.195.5408-232
* **Rust toolchain version:** nightly-x86_64-pc-windows-msvc
* **IDE name and version:** CLion
* **Operating system:** Windows10

## Problem description

Generic name and function name conflict, say constant provided when a type was expected.

![image](https://github.com/intellij-rust/intellij-rust/assets/17541209/d337db02-338d-437b-b855-ce0e1d667297)

Only the IDE has this error, the compiler will not report an error.

## Steps to reproduce

```rust
pub fn char() -> Vec<char> {
    vec!['c']
}
```

**Comment by Maksim20023** [comment]
Hi
Thank you for your report. This is related to false-positive trigger  of "Constant provided when a type was expected [E0747]" and it is already known to us.
You can subscribe to the [Github 9862](https://github.com/intellij-rust/intellij-rust/issues/9862) issue to stay informed and vote (click the emoji button in the bottom-left corner of the GitHub post).
If you have any further suggestions, please feel free to reach out.

**Comment by oovm** [comment]
closed for duplicate
